modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/ExecutionSpan.java

```diff
@@ -165,7 +168,9 @@ public Long getEnd() {
 
   @Override
   public void addAttribute(String key, String value) {
-    additionalAttributes.put(key, value);
+    if (!key.equals(SPAN_KIND) && !key.equals(STATUS)) {
```

Comment:
Put a TODO here indicating: This should be moved to the OpentelemetrySpanExporter foreach (it only makes sense in optel) after performance assessment. 
https://github.com/mulesoft/mule/blob/master/modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/impl/exporter/OpenTelemetrySpanExporter.java#L134

I've created a ticket for the TODO W-12297079

Reply:
I added this suggested change on this PR